juju/juju
# Official build

## Description of change

This adds necessary changes to https://github.com/juju/juju/pull/7920 (see the last commit for the changes).
It renames the version file to juju-versions.yaml and ensure upgrade-juju works as needed. It used to just inspect the version of the running agent to determine if implicit upload was needed but it now needs to check the local jujud version as well.

## QA steps

bootstrap using a local jujud with a versions file indicating it is an official build
change the jujud
$ juju upgrade-juju 
it should upgrade with an incremented build number
